source/retryable-writes/retryable-writes.rst

```diff
@@ -441,22 +453,20 @@ Consider the following pseudo-code:
      * values will be derived from the implicit or explicit session object. */
     retryableCommand = addTransactionIdToCommand(command, session);
 
-    /* NetworkException and NotMasterException are both retryable errors. If
-     * caught, remember the exception, update SDAM accordingly, and proceed with
-     * retrying the operation.
+    /* If the error has a RetryableWriteError label, remember the exception
+     * and proceed with retrying the operation.
      *
      * IllegalOperation (code 20) with errmsg starting with "Transaction
      * numbers" MUST be re-raised with an actionable error message. */
     try {
       return executeCommand(server, retryableCommand);
-    } catch (NetworkException originalError) {
-      updateTopologyDescriptionForNetworkError(server, originalError);
-    } catch (NotMasterException originalError) {
-      updateTopologyDescriptionForNotMasterError(server, originalError);
-    } catch (IllegalOperation originalError) {
-      if ( originalError.code == 20 && originalError.errmsg.startsWith("Transaction numbers") ) {
-        originalError.errmsg = "This MongoDB deployment does not support retryable...";
-        throw originalError;
+    } catch (Exception originalError) {
+      handleError(e);
+      if (!e.hasErrorLabel("RetryableWriteError")) {
```

Comment:
the psuedocode hadn't actually been updated yet to include the newly specified label-based behavior, so I did so here. No changes specific to `PoolClearedError` were really necessary as a result.

Reply:
Oh, is this now covered by `handleError`? I just saw that below, and the reference to SDAM.